crates/astria-sequencer/src/ibc/ics20_transfer.rs

```diff
@@ -365,14 +370,25 @@ impl AppHandlerExecute for Ics20Transfer {
     ) -> anyhow::Result<()> {
         use penumbra_ibc::component::packet::WriteAcknowledgement as _;
 
+        let use_new_error_format = state
+            .get_upgrade_change_info(&Aspen::NAME, &IbcAcknowledgementFailureChange::NAME)
+            .await
+            .map_err(|err| eyre_to_anyhow(err).context("failed to read upgrade info"))?
+            .is_some();
+
         let ack = match receive_tokens(&mut state, &msg.packet).await {
             Ok(()) => TokenTransferAcknowledgement::success(),
             Err(e) => {
                 tracing::debug!(
```

Comment:
I think this should be upgraded to a `warn!` if the full error is no longer available elsewhere (in case of `use_new_error_format`)

Reply:
Done in [6f4a27c](https://github.com/astriaorg/astria/pull/2101/commits/6f4a27ca05b8311b371458101147d6fd9b710c7c).